pages/index.js

```diff
@@ -56,13 +61,17 @@ export default function HomePage() {
         {submittedData.length > 0 && (
           <div className="submitted-data">
             {submittedData.map((data, index) => (
-              <p key={index}>
-                Was ist zu tun: {data.option} <br />
-                Bis wann erledigt: {data.date}
-              </p>
+              <div key={index} className="task-item">
+                <p>
+                  Was ist zu tun: {data.option} <br />
+                  Bis wann erledigt: {data.date}
+                </p>
+                <button onClick={() => handleDelete(index)}>Delete</button>
```

Comment:
Do not delete using the index, instead use data.id. Also use the id for the `key` attribute. If you do not have an id in you data, make sure to create one using nanoid, uid or something similar.

```suggestion
                <button onClick={() => handleDelete(data.id)}>Delete</button>
```